tsl/src/nodes/decompress_chunk/compressed_batch.c

```diff
@@ -18,29 +18,24 @@
 #include "nodes/decompress_chunk/compressed_batch.h"
 #include "nodes/decompress_chunk/vector_predicates.h"
 
-/*
- * Create a single value ArrowArray from Postgres Datum. This is used to run
- * the usual vectorized predicates on compressed columns with default values.
- */
 static ArrowArray *
-make_single_value_arrow(Oid pgtype, Datum datum, bool isnull)
+make_single_value_arrow_arithmetic(Oid arithmetic_type, Datum datum, bool isnull)
```

Comment:
Could you add a comment about the high-level difference between `make_single_value_arrow_arithmetic` and `make_single_value_arrow_text`? It might help understand the code better without comparing the details for the functions.

Reply:
They are different just because one creates arrow array for arithmetic types, and the other for text types, and the layouts of these arrays are different. I can write "this function creates arrow array for arithmetic types", but this is already clear from the name, so tell me if you have better ideas for a comment.

Reply:
I guess if you add a comment and just mention that the array layouts are different would be enough for the reader to know at this point.